crates/router/src/core/payment_methods/cards.rs

```diff
@@ -954,6 +960,28 @@ pub async fn list_payment_methods(
                 banks_consolidated_hm.insert(element.payment_method_type, vec![connector]);
             }
         }
+
+        if element.payment_method == api_enums::PaymentMethod::BankRedirect {
+            let connector = element.connector.clone();
+            if let Some(vector_of_connectors) =
+                banks_consolidated_hm.get_mut(&element.payment_method_type)
+            {
+                vector_of_connectors.push(connector);
+            } else {
+                banks_consolidated_hm.insert(element.payment_method_type, vec![connector]);
+            }
+        }
```

Comment:
This code seems to be the same as code from line 953 to 962. Was this accidentally committed.

Reply:
Also can this be broken down into smaller functions as there is too much nesting. `#[inline]` can be done on those smaller functions.